### ISSUE TITLE
Header: All menu navigation links should appear in the hamburger dropdown

### ISSUE THREAD

**Issue by alundgard** [sul-dlss/exhibits#2772]
Currently, only the top-level navigation links appear in the hamburger. The lower-level links appear below the title.
https://exhibits.stanford.edu/martin-wong
<img width="486" alt="Image" src="https://github.com/user-attachments/assets/31288ec4-1d5f-4353-acbc-9df37cd37014" />

We'd like all of the navigation links to appear in the hamburger. As an example, see the [Archives implementation](https://archives.stanford.edu/collections).
<img width="485" alt="Image" src="https://github.com/user-attachments/assets/7818a9ef-58ff-4be8-a032-d2d0626604d0" />

This is already implemented in the component library. See example: https://sul-dlss.github.io/component-library/header/.
<img width="486" alt="Image" src="https://github.com/user-attachments/assets/5bb73bd3-c712-4917-8b27-7eeeed05dc80" />

**Comment by dnoneill** [comment]
@alundgard Should the mobile menu also open under the title or should it appear as it is?

**Comment by alundgard** [comment]
Yes there is a separate issue for that: https://github.com/sul-dlss/component-library/issues/39.

**Comment by dnoneill** [comment]
I would wait for https://github.com/sul-dlss/component-library/pull/86 to be merged (there could be changes after review) before addressing this ticket. The code should be easy to implement once it is in the component library.
